src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -119,6 +120,15 @@ private void processZipFile(CloudBlobContainer container, String zipFilename)
             cloudBlockBlob.delete();
 
             envelopeProcessor.markAsProcessed(envelope, container.getName(), zipFilename);
+        } catch (DuplicateEnvelopeException exception) {
+            envelopeProcessor.markAsDuplicate(
+                exception.getMessage(),
+                exception.getEnvelope(),
+                container.getName(),
```

Comment:
We already have the envelope so no need to pass container name as it can be derived from Envelope.

Reply:
Yea.. didn't change current structure of all those _markers_. This will be completely revised when error handler comes in to play